x-pack/legacy/plugins/siem/server/client/client.ts

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Legacy } from 'kibana';
+
+import { APP_ID, SIGNALS_INDEX_KEY } from '../../common/constants';
+
+export class SiemClient {
+  public signalsIndex: string;
```

Comment:
I'd be careful with too many classes or factories as they can have bad side effects and state management can be problematic.

These are good clean classes with great tests you have written, but still, I would just be cautious as the next dev showing up might abuse these later. I typically avoid classes altogether to hold state and just use normal objects when I can.

With that said, for this, I think you want to create immutable classes with mostly private members that cannot be changed after construction. For example having this with a public string makes it easy for me to change its value in different parts of the application after its initialized with a different signals index and then other parts of the application with a reference will end up having a different signals index.

I would use this to make it immutable and only changeable in the constructor:

```ts
public readonly signalsIndex: string;
```

And do the same anywhere else with `readonly`

Reply:
https://github.com/elastic/kibana/pull/58292/commits/695841f0c1c5db6fac373c02f139026a61a9a83a